platform/kubernetes/resource/spec.go

```diff
@@ -0,0 +1,106 @@
+package resource
+
+import (
+	"fmt"
+
+	"github.com/weaveworks/flux/diff"
+)
+
+// Types that daemonsets, deployments, and other things have in
```

Comment:
I'm worried that we're coupling ourselves to the k8s API here. I.e. we can't support older clusters (e.g. they don't have annotations) and we'd have to update if they add anything.